src/aframe_components/custom-components.ts

```diff
@@ -50,36 +52,41 @@ export const guideComponent: Partial<Guidance> = {
    */
   tick(time, timeDelta) {
     this.state.currPosition = this.el.object3D.position;
```

Comment:
Have you considered, instead of mutating the state, building a new state object in this function and assigning it at the end to this.state?  I get nervous when I see a lot of object mutations because it can lead to very difficult to track bugs, especially if you are mutating from different methods etc.